sbi/samplers/vi/vi_pyro_flows.py

```diff
@@ -0,0 +1,635 @@
+    """Registers a function that builds a normalizing flow.
+
+    Args:
+        cls: Builder that is registered.
+        name: Name of the builder.
+
+
+    """
+
+    def _register(cls):
+        if name is None:
+            cls_name = cls.__name__
+        else:
+            cls_name = name
+        if cls_name in _FLOW_BUILDERS:
+            raise ValueError(f"The flow {cls_name} is not registered as default.")
+        else:
+            _FLOW_BUILDERS[cls_name] = cls
+        return cls
+
+    if cls is None:
+        return _register
+    else:
+        return _register(cls)
+
+
+def get_default_flows() -> List[str]:
+    """Returns names of all registered flow builders.
+
+    Returns:
+        List[str]: List of names.
+
+    """
+    return list(_FLOW_BUILDERS.keys())
+
+
+def get_flow_builder(
+    name: str,
+    event_shape: torch.Size,
+    link_flow: TorchTransform,
+    device: str = "cpu",
+    **kwargs,
+) -> Distribution:
+    """Returns an normalizing flow, by instantiating the flow build with all arguments.
+
+    Args:
+        name: Name of the flow.
+        event_shape: Event shape.
+        link_flow: Transform that maps to the prior support.
+        device: Device on which to build.
+
+    Returns:
+        Distribution: Builded trainable distribution.
+
+    """
+    builder = _FLOW_BUILDERS[name]
+    return builder(event_shape, link_flow, device=device, **kwargs)
+
+
+# Initialization functions.
+
+
+def init_affine_autoregressive(dim: int, device: str = "cpu", **kwargs):
+    """Provides the default initial arguments for an affine autoregressive transform."""
+    hidden_dims = kwargs.pop("hidden_dims", [5 * dim + 5])
+    skip_connections = kwargs.pop("skip_connections", False)
+    nonlinearity = kwargs.pop("nonlinearity", nn.ReLU())
+    arn = AutoRegressiveNN(
+        dim, hidden_dims, nonlinearity=nonlinearity, skip_connections=skip_connections
+    ).to(device)
+    return [arn], {"log_scale_min_clip": -3.0}
+
+
+def init_spline_autoregressive(dim, device: str = "cpu", **kwargs):
+    """Provides the default initial arguments for an spline autoregressive transform."""
+    hidden_dims = kwargs.pop("hidden_dims", [5 * dim + 5])
+    skip_connections = kwargs.pop("skip_connections", False)
+    nonlinearity = kwargs.pop("nonlinearity", nn.ReLU())
+    count_bins = kwargs.get("count_bins", 10)
+    order = kwargs.get("order", "linear")
+    bound = kwargs.get("bound", 5)
+    if order == "linear":
+        param_dims = [count_bins, count_bins, (count_bins - 1), count_bins]
+    else:
+        param_dims = [count_bins, count_bins, (count_bins - 1)]
+    neural_net = AutoRegressiveNN(
+        dim,
+        hidden_dims,
+        param_dims=param_dims,
+        skip_connections=skip_connections,
+        nonlinearity=nonlinearity,
+    ).to(device)
+    return [dim, neural_net], {"count_bins": count_bins, "bound": bound, "order": order}
+
+
+def init_affine_coupling(dim, device: str = "cpu", **kwargs):
```

Comment:
I think we can remove this one, no?

Reply:
The coupling flows. I guess the only advantage is that the forward and inverse passes are equally expensive, which may help in high dimensions. But yeah I guess we can remove it.

Reply:
I'm also ok with keeping them. No strong opinions here

Reply:
Yeah, lets keep them.